fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/IndexingByRecords.java

```diff
@@ -75,30 +75,32 @@
     @Nonnull private static final byte[] END_BYTES = new byte[]{(byte)0xff};
 
     @Nonnull private final TupleRange recordsRange;
-    @Nonnull private final byte[] myTypeStamp;
+    @Nonnull private final IndexBuildProto.IndexBuildIndexingStamp myIndexingTypeStamp;
 
     IndexingByRecords(@Nonnull IndexingCommon common) {
         super(common);
         this.recordsRange = computeRecordsRange();
-        this.myTypeStamp = compileTypeStamp();
+        this.myIndexingTypeStamp = compileIndexingTypeStamp();
     }
 
     @Override
     @Nonnull
-    byte[] getTypeStamp() {
-        return myTypeStamp;
+    IndexBuildProto.IndexBuildIndexingStamp getIndexingTypeStamp() {
+        return myIndexingTypeStamp;
     }
 
     @Nonnull
-    byte[] compileTypeStamp() {
-        return IndexMetaDataProto.IndexMetaDataIndexingStamp.newBuilder()
-                .setMethod(IndexMetaDataProto.IndexMetaDataIndexingStamp.Method.BY_RECORDS)
-                .build().toByteArray();
+    private static IndexBuildProto.IndexBuildIndexingStamp compileIndexingTypeStamp() {
+        return
+                IndexBuildProto.IndexBuildIndexingStamp.newBuilder()
+                        .setMethod(IndexBuildProto.IndexBuildIndexingStamp.Method.BY_RECORDS)
+                        .build();
```

Comment:
I think this message can be a `static final` constant for the whole class, as all instances will have the same message.